modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -44,76 +44,145 @@ class Instrument_Manager extends \NDB_Menu_Filter
     {
         $config        = \NDB_Config::singleton();
         $this->path    = $config->getSetting("base");
-        $this->headers = array(
+        $this->headers = [
                           'Instrument',
                           'Instrument_Type',
                           'Table_Installed',
                           'Table_Valid',
                           'Pages_Valid',
-                         );
+                         ];
         $this->query   = " FROM test_names t";
-        $this->columns = array(
+        $this->columns = [
                           't.Test_name as Instrument',
                           "'x' as Instrument_Type",
                           "'x' as Table_Installed",
                           "'x' as Table_Valid",
                           "'x' as Pages_Valid",
-                         );
-        $this->validFilters = array('');
-        $this->formToFilter = array();
+                         ];
+        $this->validFilters = [''];
+        $this->formToFilter = [];
 
-        $writable = true;
-        if (!is_writable($this->path . "project/instruments")
-            || !is_writable($this->path . "project/tables_sql")
-        ) {
-            $writable = false;
-        } else {
-            $writable = true;
+        // Check to see whether the quatUser is configured.
+        // If this user is not configured, the instrument cannot be
+        // automatically installed as the table will not be able to be
+        // created.
+        $quat_user_is_configured = $this->isQuatUserConfigured();
+        if (!$quat_user_is_configured) {
+            $this->tpl_data['can_install'] = false;
+            $this->tpl_data['feedback']    = "Instrument installation is not "
+                . "possible given the current server configuration. You can "
+                . "upload files but they will have to be installed by your "
+                . "LORIS administrator.";
         }
+
+        // Ensure we can write to folders in project/. If not, the upload
+        // form will not be created on the front-end. Warnings will explain to
+        // to the user that installation is not possible.
+        $writable = is_writable($this->path . "project/instruments")
+            && is_writable($this->path . "project/tables_sql");
         $this->tpl_data['writable'] = $writable;
+        if (!$writable) {
+            $this->tpl_data['feedback'] = "Instrument installation has been "
```

Comment:
This is a assumption. The fact is that the directory is not writable for the webserver user.

Reply:
Yes but it is consistent with how this module worked before. I explain this in the PR description.

Reply:
The previous error message in the smarty template was "-Instrument directory or tables_sql not writable.
-Automatic uploading of instruments has been disabled.", which doesn't make that assumption.

Reply:
Ah see I read that as "Automatic uploading of instruments has been disabled [by your administrator]" not disabled as a result of the misconfiguration. 

I'll correct the message here.